packages/client/src/views/VerifyCertificate/VerifyCertificatePage.tsx

```diff
@@ -0,0 +1,551 @@
+`
+
+const SpinnerContainer = styled.div`
+  position: relative;
+`
+
+const StyledSpinner = styled(Spinner)`
+  position: absolute;
+  margin-left: -24px;
+  margin-top: -24px;
+  top: calc(50% - 20px);
+  left: 50%;
+  width: 30px;
+  height: 30px;
+`
+// fetchRecordDetailsForVerification
+const FETCH_RECORD_DETAILS_FOR_VERIFICATION = gql`
+  query fetchRecordDetailsForVerification($id: String!) {
+    fetchRecordDetailsForVerification(id: $id) {
+      ... on BirthRegistration {
+        id
+        child {
+          name {
+            firstNames
+            familyName
+          }
+          birthDate
+          gender
+        }
+        eventLocation {
+          name
+          description
+          type
+          address {
+            city
+            districtName
+            stateName
+          }
+        }
+        registration {
+          trackingId
+          registrationNumber
+          type
+        }
+        createdAt
+        history {
+          action
+          regStatus
+          user {
+            name {
+              firstNames
+              familyName
+            }
+            catchmentArea {
+              name
+            }
+          }
+        }
+      }
+      ... on DeathRegistration {
+        id
+        deceased {
+          name {
+            firstNames
+            familyName
+          }
+          birthDate
+          gender
+          deceased {
+            deathDate
+          }
+        }
+        eventLocation {
+          name
+          description
+          type
+          address {
+            city
+            districtName
+            stateName
+          }
+        }
+        registration {
+          trackingId
+          registrationNumber
+          type
+        }
+        createdAt
+        history {
+          action
+          regStatus
+          user {
+            name {
+              firstNames
+              familyName
+            }
+            catchmentArea {
+              name
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+export function VerifyCertificatePage() {
+  const intl = useIntl()
+  const { declarationId } = useParams<{ declarationId: string }>()
+
+  const logo = useSelector(selectCountryLogo)
+  const appName = useSelector(selectApplicationName)
+
+  const [closeWindow, setCloseWindow] = React.useState(false)
+  const [timeOut, setTimeOut] = React.useState(false)
+
+  const { loading, error, data } = useQuery(
+    FETCH_RECORD_DETAILS_FOR_VERIFICATION,
+    {
+      variables: { id: declarationId },
+      fetchPolicy: 'network-only'
+    }
+  )
+
+  const [currentData, setCurrentData] = React.useState<
+    BirthRegistration | DeathRegistration
+  >()
+
+  React.useEffect(() => {
+    if (data) {
+      setTimeout(() => {
+        setCloseWindow(true)
+      }, 60000) // 60000 is correct value
+
+      setTimeout(() => {
+        setCloseWindow(false)
+        setTimeOut(true)
+      }, 600000) // 600000 is correct value
```

Comment:
We need to write cleanups for these timeouts

Reply:
@wilrona so in effect, you should store the return values of the setTimeouts and return a new function from this useEffect. The returning function should call `clearTimeout(timeoutId)` for all the setTimeout return values where `timeoutId` is the return value.

Reply:
We could also create something similar to

```ts
export const useTimeout = (callback: () => void, timeout = 0) => {
  useEffect(() => {
    const timeoutReference = setTimeout(callback, timeout);
 
    return () => clearTimeout(timeoutReference);
   }, [])
}
```

```ts
useTimeout(() => { setCloseWindow(true) }, 60000)
useTimeout(() => { setTimeOut(true) }, 600000)
```

Reply:
OK thank you